core/src/executor/evaluate/function.rs

```diff
@@ -274,6 +276,20 @@ pub fn floor(name: String, n: Evaluated<'_>) -> Result<Value> {
     Ok(Value::F64(eval_to_float!(name, n).floor()))
 }
 
+pub fn rand(n: Option<Evaluated<'_>>) -> Result<Value> {
+    let mut rng = StdRng::seed_from_u64(1_u64); // fix me!! this needs to be a global var.
+
+    match n {
+        Some(x) => match x.try_into()? {
+            Value::I8(v) => rng = StdRng::seed_from_u64(v as u64),
+            Value::I64(v) => rng = StdRng::seed_from_u64(v as u64),
+            _ => {}
+        },
+        None => {}
+    }
+    Ok(Value::F64(rng.gen()))
```

Comment:
This looks ok with seed, but without seed, it needs not to be deterministic.

```rust
let v = match seed {
    Some(seed) => seed.try_into()?,
    None => {
        return Ok(Value::F64(rand::random()));
    }
};  
    
let v = match v {
    Value::I8(v) => StdRng::seed_from_u64(v as u64).gen(),
    Value::I64(v) => StdRng::seed_from_u64(v as u64).gen(),
    _ => rand::random(), // may be we need to return error?
};  

Ok(Value::F64(v))
```

Reply:
One issue, I am having..

error[E0308]: mismatched types
  --> core/src/plan/expr/function.rs:39:26
   |
17 |         match self {
   |               ---- this expression has type `&ast::function::Function`
18 |             Self::Now() | Function::Pi() | Function::GenerateUuid() => Exprs::Empty(empty()),
19 |             Self::Lower(expr)
   |                         ---- first introduced with type `&ast::expr::Expr` here
...
39 |             | Self::Rand(expr)
   |                          ^^^^ expected enum `ast::expr::Expr`, found enum `std::option::Option`
   |
Do we need to create another function similar to Exprs::Single([expr].into_iter()),, that accepts Optional arguments or is there a way to make it work with the current Exprs::Single?